curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/write-higher-order-arrow-functions.english.md

```diff
@@ -1,42 +1,73 @@
+})<br>
+console.log(results)<br>
+// => [ {thumbnail: "someIcon", likes: 432, shares: 600}, {thumbnail: "Another icon", likes: 300, shares: 501}]
+    </blockquote>
+
+Find more info about [arrow-functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions).<br>
+And about higher-order functions like [forEach](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach), [map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map), and [filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter).
+
 </section>
 
+
+
 ## Instructions
 <section id='instructions'>
-Use arrow function syntax to compute the square of only the positive integers (decimal numbers are not integers) in the array <code>realNumberArray</code> and store the new array in the variable <code>squaredIntegers</code>.
+Use arrow function syntax to compute the square of <em>only</em> the positive integers (decimal numbers are not integers) in the array <code>realNumberArray</code> and store the new array in the variable <code>squaredIntegers</code>.
 </section>
 
 ## Tests
 <section id='tests'>
 
 ```yml
 tests:
-  - text: <code>squaredIntegers</code> should be a constant variable (by using <code>const</code>).
+  - text: <code>squareList</code> should be a <code>function</code>.
+    testString: assert.typeOf(squareList, 'function'), '<code>squareList</code> should be a <code>function</code>';
+  - text: <code>squareList</code> should be a constant variable (by using         <code>const</code>).
     testString: getUserInput => assert(getUserInput('index').match(/const\s+squaredIntegers/g), '<code>squaredIntegers</code> should be a constant variable (by using <code>const</code>).');
-  - text: <code>squaredIntegers</code> should be an <code>array</code>
-    testString: assert(Array.isArray(squaredIntegers), '<code>squaredIntegers</code> should be an <code>array</code>');
-  - text: <code>squaredIntegers</code> should be <code>[16, 1764, 36]</code>
-    testString: assert.deepStrictEqual(squaredIntegers, [16, 1764, 36], '<code>squaredIntegers</code> should be <code>[16, 1764, 36]</code>');
   - text: <code>function</code> keyword was not used.
     testString: getUserInput => assert(!getUserInput('index').match(/function/g), '<code>function</code> keyword was not used.');
   - text: loop should not be used
     testString: getUserInput => assert(!getUserInput('index').match(/(for)|(while)/g), 'loop should not be used');
   - text: <code>map</code>, <code>filter</code>, or <code>reduce</code> should be used
     testString: getUserInput => assert(getUserInput('index').match(/map|filter|reduce/g), '<code>map</code>, <code>filter</code>, or <code>reduce</code> should be used');
+  - text: <code>Function should return an <code>array</code> called <code>squaredIntegers</code>
```

Comment:
Could you change the text here a bit?:

```
- text: <code>squareList should return an <code>array</code> called <code>squaredIntegers</code>
```

Or possibly:

```
- text: The function <code>squareList should return an <code>array</code> called <code>squaredIntegers</code>
```

Reply:
yep.